### PR TITLE
Migrate to pydantic-v2

### PR DESCRIPTION
**Pull request by wu-clan**
WIP.

### CODE REVIEW

**Comment by wu-clan** [comment]
Need to wait for #253 to merge and resolve the conflict. The details are shown in #180.

**Comment by wu-clan** [comment]
This PR is first merged into the pydantic-v2 branch, then renamed and locked v1.

File: backend/app/common/rbac.py [hunk 1]
```diff
@@ -59,6 +59,7 @@ async def rbac_verify(self, request: Request, _: dict = DependsJwtAuth) -> None:
         method = request.method
         if settings.MENU_PERMISSION:
             # 菜单权限校验
+            # TODO: 改用流行方案，自定义接口权限字段标识
```
Comment (downdawn) [inline backend/app/common/rbac.py:62]: Excited

File: backend/app/schemas/user.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ class RegisterUser(Auth):
     nickname: str | None = None
     email: str = Field(..., example='user@example.com')
 
-    @validator('email')
+    @field_validator('email')
     def email_validate(cls, v):
         try:
             validate_email(v, check_deliverability=False).email
```
Comment (downdawn) [inline backend/app/schemas/user.py:30]: [https://docs.pydantic.dev/latest/api/networks/#pydantic.networks.EmailStr](https://docs.pydantic.dev/latest/api/networks/#pydantic.networks.EmailStr)

File: backend/app/schemas/user.py [hunk 3]
```diff
@@ -55,15 +56,17 @@ class _UserInfoBase(SchemaBase):
...
         except EmailNotValidError:
             raise ValueError('邮箱格式错误')
         return v
 
-    @validator('phone')
+    @field_validator('phone')
+    @classmethod
     def phone_validate(cls, v):
```
Comment (downdawn) [inline backend/app/schemas/user.py:70]: [https://docs.pydantic.dev/latest/api/pydantic_extra_types_phone_numbers/](https://docs.pydantic.dev/latest/api/pydantic_extra_types_phone_numbers/)

**Review by downdawn — APPROVED** [review]

File: backend/app/schemas/user.py (hunk 2 above)
Comment (wu-clan) [reply backend/app/schemas/user.py:30]: I tried `EmailStr` in v1, but I forgot the reason why I didn't use it. I'll try again.

Comment (wu-clan) [reply backend/app/schemas/user.py:30]: updated.

File: backend/app/schemas/user.py (hunk 3 above)
Comment (wu-clan) [reply backend/app/schemas/user.py:70]: updated

Comment (wu-clan) [reply backend/app/schemas/user.py:70]: updated.

**Pull request merged**